infra/resources/modules/commons/function_async.tf

```diff
@@ -154,6 +154,51 @@ resource "azurerm_cosmosdb_sql_role_assignment" "subs_async_fn_to_cosmos_db" {
   principal_id        = module.subscription_async_fn.system_identity_principal
 }
 
+# Enables the subscription_async_fn to create user assigned identity
+resource "azurerm_role_assignment" "subs_asyn_create_user_assigned_identity" {
+  scope                = azurerm_resource_group.data_rg.id
+  role_definition_name = "Managed Identity Contributor"
+  principal_id         = module.subscription_async_fn.system_identity_principal
+}
+
+# Create queue
+resource "azurerm_role_definition" "subs_asyn_create_queue" {
+  name        = "Service Bus Create Queue"
+  scope       = azurerm_servicebus_namespace.main.id
```

Comment:
Shouldn't it be the queue id rather than the namespace id?

Reply:
I'm not sure of that.
The role definition should give permission to the async function to create queues in a specific namespace (which is the service bus' namespace).

Moreover, the queue is created by the application, so in terraform is not possible to reference it.